openedx/core/lib/cache_utils.py

```diff
@@ -1,12 +1,50 @@
 """
 Utilities related to caching.
 """
+import collections
 import cPickle as pickle
 import functools
 import zlib
 from xblock.core import XBlock
 
 
+class Memoized(object):
+    """
+    Decorator. Caches a function's return value each time it is called.
+    If called later with the same arguments, the cached value is returned
+    (not reevaluated).
+
+    https://wiki.python.org/moin/PythonDecoratorLibrary#Memoize
+    """
+    def __init__(self, func):
+        self.func = func
+        self.cache = {}
```

Comment:
This `memoize` is meant to be used for data that is constant throughout the lifetime of a gunicorn worker process, possibly costly to compute, and that is required often.  Yes, it needs to be used wisely.

However, after speaking with @cpennington, I decided to move the memoizing to the XBlock.load_classes method instead (in the XBlock repo) so its benefit can be used more widely.